gwt/pom.xml

```diff
@@ -40,39 +40,10 @@
       <classifier>sources</classifier>
       <version>${project.version}</version>
     </dependency>
+    <dependency>
+      <groupId>javax.inject</groupId>
+      <artifactId>javax.inject</artifactId>
+      <classifier>sources</classifier>
```

Comment:
And yes, GWT needs both .class and .java files (for annotations, for other classes it only needs .java files, but you'll run JavaC on your code 99.9% of the time, so you also need .class files anyway)